test/e2e/flask/petnames-transactions.spec.js

```diff
@@ -33,15 +45,66 @@ describe('Petnames', function () {
       async ({ driver }) => {
         await unlockWallet(driver);
         await openDapp(driver);
-        await createTransactionRequest(driver);
+        await createTransactionRequestInDapp(driver);
         await switchToNotificationWindow(driver, 3);
         await expectName(driver, '0x0c54F...7AaFb', false);
-        await saveName(driver, '0x0c54F...7AaFb', 'test.lens', undefined);
+
+        // Test custom name.
+        await saveName(driver, '0x0c54F...7AaFb', 'Custom Name', undefined);
         await rejectSignatureOrTransactionRequest(driver);
         await focusTestDapp(driver);
-        await createTransactionRequest(driver);
+        await createTransactionRequestInDapp(driver);
         await switchToNotificationWindow(driver, 3);
-        await expectName(driver, 'test.lens', true);
+        await expectName(driver, 'Custom Name', true);
+
+        // Test proposed name.
+        await saveName(driver, 'Custom Name', undefined, 'test4.lens', true);
+        await rejectSignatureOrTransactionRequest(driver);
+        await focusTestDapp(driver);
+        await createTransactionRequestInDapp(driver);
+        await switchToNotificationWindow(driver, 3);
+        await expectName(driver, 'test4.lens', true);
+      },
+    );
+  });
+
+  it('can update custom names for addresses in transaction requests', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withPreferencesController({
+            featureFlags: {
+              sendHexData: true,
+            },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+        createTransactionRequestInWallet(
+          driver,
+          '0x0c54fccd2e384b4bb6f2e405bf5cbc15a017aafb',
+        );
+
+        // Test custom name.
+        await saveName(driver, '0x0c54F...7AaFb', 'Custom Name', undefined);
+        await rejectSignatureOrTransactionRequest(driver);
+        await createTransactionRequestInWallet(
+          driver,
+          '0x0c54fccd2e384b4bb6f2e405bf5cbc15a017aafb',
+        );
+        await expectName(driver, 'Custom Name', true);
+
+        // Test proposed name.
+        await saveName(driver, 'Custom Name', undefined, 'test4.lens', true);
+        await rejectSignatureOrTransactionRequest(driver);
+        await createTransactionRequestInWallet(
+          driver,
+          '0x0c54fccd2e384b4bb6f2e405bf5cbc15a017aafb',
```

Comment:
Is it worth using some constants at the top of this file for the address, truncated address, and names?

Reply:
Done.